### PR TITLE
BOHB sanity checks

### PR DESCRIPTION
**Pull request by Yard1**
This PR fixes issues with the special BOHB scheduler not being set properly when using BOHB, as well as prevents the user from using it with other search algorithms.

### CODE REVIEW

File: tune_sklearn/tune_search.py [hunk 1]
```diff
@@ -314,10 +314,24 @@ def __init__(self,
...
             import ConfigSpace as CS
             can_use_param_distributions = isinstance(check_param_distributions,
                                                      CS.ConfigurationSpace)
+            if early_stopping is False:
+                raise ValueError(
+                    "early_stopping must not be False when using BOHB")
+            elif not isinstance(early_stopping, HyperBandForBOHB):
+                if early_stopping != "HyperBandForBOHB":
+                    warnings.warn(
+                        "Ignoring early_stopping value, as BOHB requires HyperBandForBOHB"
```
Comment (richardliaw) [inline tune_sklearn/tune_search.py:328]:
```suggestion
                        "Ignoring early_stopping value, as BOHB requires HyperBandForBOHB as the EarlyStopping scheduler"
```

File: tune_sklearn/tune_search.py [hunk 2]
```diff
@@ -314,10 +314,24 @@ def __init__(self,
...
+                    )
+                early_stopping = "HyperBandForBOHB"
+        elif early_stopping == "HyperBandForBOHB" or isinstance(
+                early_stopping, HyperBandForBOHB):
+            raise ValueError(
+                "early_stopping can be HyperBandForBOHB only when using BOHB")
```
Comment (richardliaw) [inline tune_sklearn/tune_search.py:334]: `search_optimization must be set to 'BOHB' if early_stopping is set to HyperBandForBOHB.`

**Review by richardliaw — COMMENTED** [review]
left some small comments on the bohb error messages, otherwise looks great.

**Comment by Yard1** [comment]
Thanks, added those.

**Comment by richardliaw** [comment]
Awesome; lint is still failing though?

**Comment by Yard1** [comment]
@richardliaw Should be good now

**Review by richardliaw — APPROVED** [review]

**Pull request merged**